collector/smb_server_shares.go

```diff
@@ -0,0 +1,396 @@
+	"github.com/prometheus/common/log"
+	"gopkg.in/alecthomas/kingpin.v2"
+)
+
+func init() {
+	registerCollector("smb_server_shares", NewSmbServerSharesCollector, "SMB Server Shares")
+}
+
+var (
+	shareWhitelist = kingpin.Flag(
+		"collector.smb_server_shares.share-whitelist",
+		"Regexp of shares to whitelist. Share name must both match whitelist and not match blacklist to be included.",
+	).Default(".+").String()
+	shareBlacklist = kingpin.Flag(
+		"collector.smb_server_shares.share-blacklist",
+		"Regexp of shares to blacklist. Share name must both match whitelist and not match blacklist to be included.",
+	).Default("").String()
+)
+
+// A SmbServerSharesCollector is a Prometheus collector for perflib SmbServerShares metrics
+type SmbServerSharesCollector struct {
+	CurrentOpenFileCount        *prometheus.Desc
+	TotalFileOpenCount          *prometheus.Desc
+	FilesOpenedPerSec           *prometheus.Desc
+	CurrentDurableOpenFileCount *prometheus.Desc
+
+	shareWhitelistPattern *regexp.Regexp
+	shareBlacklistPattern *regexp.Regexp
+}
+
+// NewSmbServerSharesCollector ...
+func NewSmbServerSharesCollector() (Collector, error) {
+	const subsystem = "smb_server_shares"
+
+	return &SmbServerSharesCollector{
+		CurrentOpenFileCount: prometheus.NewDesc(
+			prometheus.BuildFQName(Namespace, subsystem, "current_open_file_count"),
+			"The number of file handles that are currently open in this share (SmbServerShares.CurrentOpenFileCount)",
+			[]string{"share"},
+			nil,
+		),
+		TotalFileOpenCount: prometheus.NewDesc(
+			prometheus.BuildFQName(Namespace, subsystem, "file_open_count"),
+			"The number of files that have been opened by the SMB File Server on behalf of its clients on this share since the server started. (SmbServerShares.TotalFileOpenCount)",
+			[]string{"share"},
+			nil,
+		),
+		FilesOpenedPerSec: prometheus.NewDesc(
+			prometheus.BuildFQName(Namespace, subsystem, "files_opened_total"),
+			"The rate, in seconds, at which files are being opened for the SMB File Server’s clients on this share. (SmbServerShares.FilesOpenedPerSec)",
+			[]string{"share"},
+			nil,
+		),
+		CurrentDurableOpenFileCount: prometheus.NewDesc(
+			prometheus.BuildFQName(Namespace, subsystem, "current_durable_open_file_count"),
+			"The number of durable file handles that are currently open on this share (SmbServerShares.CurrentDurableOpenFileCount)",
+			[]string{"share"},
+			nil,
+		),
+
+		shareWhitelistPattern: regexp.MustCompile(fmt.Sprintf("^(?:%s)$", *shareWhitelist)),
+		shareBlacklistPattern: regexp.MustCompile(fmt.Sprintf("^(?:%s)$", *shareBlacklist)),
+	}, nil
+}
+
+// Collect sends the metric values for each metric
+// to the provided prometheus Metric channel.
+func (c *SmbServerSharesCollector) Collect(ctx *ScrapeContext, ch chan<- prometheus.Metric) error {
+	if desc, err := c.collect(ctx, ch); err != nil {
+		log.Error("failed collecting smb_server_shares metrics:", desc, err)
+		return err
+	}
+	return nil
+}
+
+// :9432/dump?query=4352#4352
```

Comment:
What does this mean? :)

Reply:
it's the URL query to perflib (I had not found the microsoft documentation reference as u had in there from the copy&paste source file I used... the logic_drivers one)... I believe it is the class id in the array/list in perflib memory